lib/blob.js

```diff
@@ -1044,17 +1193,957 @@ Blob.prototype.leaseContainer = function leaseContainer(name, options) {
     headers['x-ms-proposed-lease-id'] = options.proposedLeaseId;
   }
 
+  utils.setConditionalHeaders(headers, {
+    ifModifiedSince: options.ifModifiedSince,
+    ifUnmodifiedSince: options.ifUnmodifiedSince
+  });
+
   return this.request('PUT', path, query, headers).then(function(response) {
     if (response.statusCode !== 200 && response.statusCode !== 201 && response.statusCode !== 202) {
       throw new Error("leaseContainer: Unexpected statusCode: " + response.statusCode);
     }
 
     var result = {
-      leaseId: response.headers['x-ms-lease-id']
+      leaseId: response.headers['x-ms-lease-id'],
+      eTag: response.headers['etag'],
+      lastModified: response.headers['last-modified']
     };
     if (response.headers['x-ms-lease-time']) {
       result.leaseTime = response.headers['x-ms-lease-time'];
     }
     return result;
   });
+};
+
+/**
+ * Creates a new block, page, or append blob, or updates the content of an existing block blob.
```

Comment:
Another solution would be to create different methods for every specific blob: putBlockBlob, putAppendBlob, putPageBlob and to make putBlob a private method (_putBlob). I committed a draft implementation of this (without changing putBlob to _putBlob, because it makes my tests to fail). What do you think? I think is more elegant and easier to use.